### DISCUSSION TITLE
Voice outputs appear to be missing the last syllable or two

### DISCUSSION THREAD

**Discussion by glendennis31** [deepgram/community#656] [General help]
### Which Deepgram product feature are you using?

Deepgram API - TTS

### Details

I am using the @deepgram/sdk.  Just in the last couple of days I noticed that all text to voice outputs are clipped just before the end of the audio. This wasn't happening before then. I haven't changed anything in terms of how the audio is written to file.  

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

const response = await deepgram.speak.request(

### If possible, please attach your code or paste it into the text box.

const { createClient } = require("@deepgram/sdk");
const fs = require("fs");
const dotenv = require ('dotenv');
dotenv.config();

// STEP 1: Create a Deepgram client with your API key
const deepgram = createClient(process.env.DEEPGRAM_API_KEY);

const getAudio = async (text, filename, socket) => {
  // STEP 2: Make a request and configure the request with options (such as model choice, audio configuration, etc.)
  //console.log('text is ' + text);
  if (!text) {return;}
  const response = await deepgram.speak.request(
    { text },
    {
      model: "aura-athena-en",
      //encoding: "linear16",
      encoding: "mp3",
      //container: "wav",
    }
  );

  // STEP 3: Get the audio stream and headers from the response
  const stream = await response.getStream();
  const headers = await response.getHeaders();
  if (stream) {
    //convert stream to an audio buffer
    const buffer = await getAudioBuffer(stream);
    // STEP 4. Write the audio buffer to a file  
    fs.writeFile("./audio/" + filename, buffer, (err) => {
      if (err) {
         console.error("Error writing audio to file:", err);
      } else {
        console.log("Audio file written to " + filename);
        socket.send(JSON.stringify({type: "URL", data: 'http://192.168.1.66:3000/audio/' + filename}));
      }
    });
  }

  if (headers) {
    //console.log("Headers:", headers);
  }
};

// helper function to convert stream to audio buffer -- ESSENTIAL
const getAudioBuffer = async (response) => {
  const reader = response.getReader();
  const chunks = [];

  while (true) {
    const { done, value } = await reader.read();
    if (done) break;

    chunks.push(value);
  }

  const dataArray = chunks.reduce(
    (acc, chunk) => Uint8Array.from([...acc, ...chunk]),
    new Uint8Array(0)
  );

  return Buffer.from(dataArray.buffer);
};

### If possible, please attach an example audio file to reproduce the issue.

[aead107a-0593-46a9-b366-221ce7b29ed4.mp3.zip](https://github.com/deepgram/community/files/14668907/aead107a-0593-46a9-b366-221ce7b29ed4.mp3.zip)

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by glendennis31** [comment]
Well you appear to have fixed it. Back to normal today.
